flocker/control/_protocol.py

```diff
@@ -201,6 +201,8 @@ def disconnected(self, connection):
         """
         self.connections.remove(connection)
 
+    # XXX this isn't node_changed anymore really, it's cluster state
+    # updates; other than naming it's pretty much the same logic though
```

Comment:
This probably does require splitting `NodeStateCommand` into two different AMP commands, though (one that accepts a `NodeManifestations` and one that accepts a `NodeApplications`)?  Although that implies the two different kinds of agents are speaking different protocols... Which maybe isn't what we want?  On the other hand, they're conveying different state (with different structure) so it's sort of difficult to make them speak the same protocol.  We could paper over it in various ways - eg make `SerializableArgument` accepts multiple types to allow instead of just one (actually it does already because the way `isinstance` works, wee).  But that's not really helping the underlying issues (of heterogeneous network comms).

Maybe papering over the difference is fine for now though.  We don't really anticipate introducing any more kinds of convergence agents any time soon and this isn't a public interface of the cluster (so we can change it however we want later).

I can almost imagine a nice fix where we make the different agents represent their different kinds of state using a unified structure (they almost do already ... json is json right??  he he he).  But, yea, probably doesn't make a practical difference at this point.